src/main/java/com/gotocompany/depot/http/request/util/JsonParserUtils.java

```diff
@@ -0,0 +1,115 @@
+package com.gotocompany.depot.http.request.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.google.gson.JsonSyntaxException;
+import com.gotocompany.depot.common.Template;
+import com.gotocompany.depot.exception.ConfigurationException;
+import com.gotocompany.depot.exception.InvalidTemplateException;
+import com.gotocompany.depot.message.ParsedMessage;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+
+
+public class JsonParserUtils {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .enable(DeserializationFeature.FAIL_ON_TRAILING_TOKENS);
+
+    public static JsonNode createJsonNode(String jsonTemplate) {
+        if (jsonTemplate.isEmpty()) {
+            throw new ConfigurationException("Json body template cannot be empty");
+        }
+        try {
+            return OBJECT_MAPPER.readTree(jsonTemplate);
+
+        } catch (JsonSyntaxException | IOException e) {
+            throw new ConfigurationException(String.format("Json body template is not a valid json. %s", e.getMessage()));
+        }
+    }
+
+    public static JsonNode parse(JsonNode jsonNode, ParsedMessage parsedMessage) {
+        switch (jsonNode.getNodeType()) {
+            case ARRAY:
+                return parseInternal((ArrayNode) jsonNode, parsedMessage);
+            case OBJECT:
+                return parseInternal((ObjectNode) jsonNode, parsedMessage);
+            case STRING:
+                return parseInternal((TextNode) jsonNode, parsedMessage);
+            case NUMBER:
+            case BOOLEAN:
+            case NULL:
+                return parseInternal(jsonNode, parsedMessage);
+            default:
+                throw new IllegalArgumentException("The provided Json type is not supported");
+        }
+    }
+
+    public static JsonNode parseInternal(ObjectNode objectNode, ParsedMessage parsedMessage) {
+        ObjectNode finalJsonObject = new JsonNodeFactory(false).objectNode();
+        for (Map.Entry<String, JsonNode> entry : objectNode.properties()) {
+            String keyString = entry.getKey();
+            TextNode keyStringNode = new TextNode(keyString);
+            JsonNode parsedKeyNode = parseInternal(keyStringNode, parsedMessage);
+            String parsedKeyString = parsedKeyNode.toString();
+            if (parsedKeyNode.getNodeType().equals(JsonNodeType.STRING)) {
+                parsedKeyString = parsedKeyString.substring(1, parsedKeyString.length() - 1);
+            }
```

Comment:
why parsing-key which is of string type needs these many transformations? Any edge case are we solving here?

Reply:
yeah the key can have template as well, so we need to handle that, but if its a string then we dont want extra \" so we need to remove that as well